tags.go

```diff
@@ -28,32 +20,67 @@ func serializeTags(name string, tags map[string]string) string {
 	const prefix = ".__"
 	const sep = "="
 
-	if len(tags) == 0 {
+	// switch len(tags) {
+	switch len(tags) {
+	case 0:
 		return name
+	case 1:
+		for k, v := range tags {
+			return name + prefix + k + sep + replaceChars(v)
+		}
+		panic("unreachable")
+	case 2:
+		var a, b tagPair
+		for k, v := range tags {
+			b = a
+			a = tagPair{k, replaceChars(v)}
+		}
+		if a.dimension > b.dimension {
+			a, b = b, a
+		}
+		return name + prefix + a.dimension + sep + a.value +
```

Comment:
Have you tried benchmarking this with a strings.Join()?

Reply:
string concatenation is faster